src/main/java/io/vertx/ext/mongo/AggregateOptions.java

```diff
@@ -0,0 +1,170 @@
+package io.vertx.ext.mongo;
+
+import java.util.Objects;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Options used to configure aggregate operations.
+ *
+ * @author <a href="mailto:nscavell@redhat.com">Nick Scavelli</a>
+ */
+@DataObject(generateConverter = true)
+public class AggregateOptions {
+  /**
+   * The default value of batchSize = 10.
+   */
+  public static final int  DEFAULT_BATCH_SIZE     = 20;
+  /**
+   * The default value of maxTime = 0.
+   */
+  public static final long DEFAULT_MAX_TIME       = 0L;
+  /**
+   * The default value of maxAwaiTime = 1000.
+   */
+  public static final long DEFAULT_MAX_AWAIT_TIME = 1000L;
+
+  private int     batchSize;
+  private long    maxTime;
+  private long    maxAwaitTime;
+  private Boolean allowDiskUse;
+
+  /**
+   * Default constructor
+   */
+  public AggregateOptions() {
+    this.batchSize = DEFAULT_BATCH_SIZE;
+    this.maxTime = DEFAULT_MAX_TIME;
+    this.maxAwaitTime = DEFAULT_MAX_AWAIT_TIME;
+  }
+
+  /**
+   * Copy constructor
+   *
+   * @param options the one to copy
+   */
+  public AggregateOptions(AggregateOptions options) {
+    this.batchSize = options.batchSize;
+    this.maxTime = options.maxTime;
+    this.maxAwaitTime = options.maxAwaitTime;
+    this.allowDiskUse = options.allowDiskUse;
+  }
+
+  /**
+   * Constructor from JSON
+   *
+   * @param options the JSON
+   */
+  public AggregateOptions(JsonObject options) {
+    this();
+    AggregateOptionsConverter.fromJson(options, this);
+  }
+
+  /**
+   * Convert to JSON
+   *
+   * @return the JSON
+   */
+  public JsonObject toJson() {
+    JsonObject json = new JsonObject();
+    AggregateOptionsConverter.toJson(this, json);
+    return json;
+  }
+
+  /**
+   * Get the specified time limit in milliseconds for processing operations on a cursor.
+   * If you do not specify a value for maxTime, operations will not time out.
```

Comment:
Is that possible given a default of 0 or does 0 mean no timeout?

Reply:
I changed the javadoc to explicitly state this.